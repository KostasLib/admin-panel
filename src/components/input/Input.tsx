import React from 'react';
import s from './Input.module.scss';

interface IProps extends React.HTMLProps<HTMLInputElement> {
  label?: string;
}

/**
 * A function component that renders a styled <input /> element
 * and an accompanying <label />, if the label prop is not omitted.
 */
export const Input: React.FC<IProps> = props => {
  const { label, ...rest } = props;
  return (
    <>
      {label && (
        <label className={s.label} htmlFor={props.name}>
          {label}
        </label>
      )}
      <input className={s.input} {...rest} />
    </>
  );
};
