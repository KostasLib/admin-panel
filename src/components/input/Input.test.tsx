import React from 'react';
import { shallow } from 'enzyme';
import { Input } from './Input';

describe('The <Input /> component', () => {
  it('renders an input and a label paired to the input based on its name prop', () => {
    const inputName = 'myInput';
    const component = shallow(
      <Input name={inputName} label="This is a label paired to myInput" />,
    );

    expect(component).toMatchSnapshot();
    expect(component.find('label').prop('htmlFor')).toBe(inputName);
  });

  it('renders only an input element if the label prop is omitted', () => {
    const component = shallow(<Input />);
    expect(component).toMatchSnapshot();
  });

  it('properly passes any props other than label to the rendered input element', () => {
    const className = 'I have replaced the previous className';
    const component = shallow(<Input className={className} label="Label" />);
    const input = component.find('input');
    
    expect(component).toMatchSnapshot();
    expect(input.prop('className')).toBe(className);
    expect(input.prop('label')).toBeUndefined();
  });
});
