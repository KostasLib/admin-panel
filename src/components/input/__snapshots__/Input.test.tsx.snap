// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <Input /> component properly passes any props other than label to the rendered input element 1`] = `
<Fragment>
  <label
    className="label"
  >
    Label
  </label>
  <input
    className="I have replaced the previous className"
  />
</Fragment>
`;

exports[`The <Input /> component renders an input and a label paired to the input based on its name prop 1`] = `
<Fragment>
  <label
    className="label"
    htmlFor="myInput"
  >
    This is a label paired to myInput
  </label>
  <input
    className="input"
    name="myInput"
  />
</Fragment>
`;

exports[`The <Input /> component renders only an input element if the label prop is omitted 1`] = `
<Fragment>
  <input
    className="input"
  />
</Fragment>
`;
